deepchecks/checks/methodology/performance_overfit.py

```diff
@@ -100,7 +100,7 @@ def plot_overfit():
         return CheckResult(result, check=self.__class__, header='Train-Test Difference Overfit',
                            display=[plot_overfit])
 
-    def add_condition_difference_not_greater_than(self, threshold: float) -> TD:
+    def add_condition_difference_not_greater_than(self: TD, threshold: float) -> TD:
```

Comment:
isn't it weird to put type hint on self?

Reply:
nope, this is called Self-Type. This PEP gives a good explaining why we need to use this - [link](https://www.python.org/dev/peps/pep-0673/)